modules/terminal/toggleterm/config.nix

```diff
@@ -6,31 +6,47 @@
 with lib;
 with builtins; let
   cfg = config.vim.terminal.toggleterm;
+  toggleKey = "<c-t>";
 in {
   config = mkIf cfg.enable {
     vim.startPlugins = [
       "toggleterm-nvim"
     ];
```

Comment:
I believe lazygit requires, well, lazygit to be installed. Perhaps we can have a module that takes `nullOr package` as a type to optionally let the user bring their own lazygit.

Reply:
You can use lib.optionals to append the package to the vim.startPlugins, which takes an input from the flake or a nixpkgs package. See `pluginsTypes` in `lib/types/plugins.nix`.

Reply:
```nix
vim.startPlugins = [
    "toggleterm-nvim"
] ++ lib.optionals (lazygit.package != null) [
    cfg.lazygit.package
];
```

out of my head, this feels like a sane approach